src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs

```diff
@@ -43,44 +44,49 @@ public static IResourceBuilder<AzureStorageResource> AddAzureStorage(this IDistr
 
         var configureConstruct = (ResourceModuleConstruct construct) =>
         {
-            var storageAccount = construct.AddStorageAccount(
-                name: name,
-                kind: StorageKind.StorageV2,
-                sku: StorageSkuName.StandardGrs
-                );
-
-            // Unfortunately Azure Storage does not list ACA as one of the resource types in which
-            // the AzureServices firewall policy works. This means that we need this Azure Storage
-            // account to have its default action set to Allow.
-            storageAccount.AssignProperty(p => p.NetworkRuleSet.DefaultAction, "'Allow'");
-
-            storageAccount.Properties.Tags["aspire-resource-name"] = construct.Resource.Name;
-
-            // Set the minimum TLS version to 1.2 to ensure resources provisioned are compliant
-            // with the pending deprecation of TLS 1.0 and 1.1.
-            storageAccount.AssignProperty(p => p.MinimumTlsVersion, "'TLS1_2'");
-
-            // Disable shared key access to the storage account as managed identity is configured
-            // to access the storage account by default.
-            storageAccount.AssignProperty(p => p.AllowSharedKeyAccess, "false");
-
-            var blobService = new BlobService(construct);
-
-            var blobRole = storageAccount.AssignRole(RoleDefinition.StorageBlobDataContributor);
-            blobRole.AssignProperty(p => p.PrincipalId, construct.PrincipalIdParameter);
-            blobRole.AssignProperty(p => p.PrincipalType, construct.PrincipalTypeParameter);
-
-            var tableRole = storageAccount.AssignRole(RoleDefinition.StorageTableDataContributor);
-            tableRole.AssignProperty(p => p.PrincipalId, construct.PrincipalIdParameter);
-            tableRole.AssignProperty(p => p.PrincipalType, construct.PrincipalTypeParameter);
-
-            var queueRole = storageAccount.AssignRole(RoleDefinition.StorageQueueDataContributor);
-            queueRole.AssignProperty(p => p.PrincipalId, construct.PrincipalIdParameter);
-            queueRole.AssignProperty(p => p.PrincipalType, construct.PrincipalTypeParameter);
-
-            storageAccount.AddOutput("blobEndpoint", sa => sa.PrimaryEndpoints.BlobUri);
-            storageAccount.AddOutput("queueEndpoint", sa => sa.PrimaryEndpoints.QueueUri);
-            storageAccount.AddOutput("tableEndpoint", sa => sa.PrimaryEndpoints.TableUri);
+            var storageAccount = new StorageAccount(name)
+            {
+                Location = construct.AddLocationParameter(),
```

Comment:
Why `AddLocationParameter`? There should always be a location parameter. Make it a property?

Reply:
I have this comment on the method:

https://github.com/dotnet/aspire/blob/1e2333561f2a387592d72c2d5896c7465cef9f1f/src/Aspire.Hosting.Azure/AzureConstructResource.cs#L150-L156

@tg-msft and I talked about this should be built into the CDK - add a location parameter if the Location isn't set.

Reply:
I think this can be removed once the Dev Feed gets updated.

Reply:
Yep this is now gone.

We always generate a Location parameter up front in the Infrastructure. And don't explicitly set the `.Location` property on any resource. The CDK's location resolver then wires up the parameter to the parameter for any unset properties.